src/System.Reflection.Metadata/tests/PortableExecutable/DebugDirectoryTests.cs

```diff
@@ -153,10 +153,12 @@ private void ValidateEmbeddedPortablePdb(PEReader reader)
             Assert.Equal(DebugDirectoryEntryType.Reproducible, entries[1].Type);
             Assert.Equal(DebugDirectoryEntryType.EmbeddedPortablePdb, entries[2].Type);
 
-            var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]);
-            var pdbReader = provider.GetMetadataReader();
-            var document = pdbReader.GetDocument(pdbReader.Documents.First());
-            Assert.Equal(@"C:\Documents.cs", pdbReader.GetString(document.Name));
+            using (var provider = reader.ReadEmbeddedPortablePdbDebugDirectoryData(entries[2]))
+            {
+                var pdbReader = provider.GetMetadataReader();
```

Comment:
> The provider is just a convenient factory.

That may be when one is aware of the implementation.  From my naive perspective as someone unaware of the implementation, though, I fully expect to see the reader keeping the thing it's reading alive, so that situations like this test don't happen.  It'd be really strange, for example, if I wrote ```new StreamReader(someStream)``` and then tried to read from the reader and had it throw an exception because it used a weak reference to hold the stream and allowed it to be GC'd.

Even if MetadataReader's ctor just takes a pointer and a length, I would expect one created by GetMetadataReader to hold onto more, e.g.
```C#
private sealed class ProviderMetadataReader : MetadataReader
{
    private readonly object _keepalive;
    internal unsafe ProviderMetadataReader(object keepalive, byte* metadata, int length) : base(metadata, length)
    {
        _keepalive = keepalive;
    }
}
...